torch/onnx/_internal/fx/passes/fx_to_onnxscript.py

```diff
@@ -125,10 +130,40 @@ def _retrieve_or_adapt_input_to_graph_set(fx_node_arg, fx_name_to_onnxscipt_valu
         # 2. fx_node_arg (variable in torch.fx.Graph) is be mapped to
         #    torch.jit.Value, fx_name_to_onnxscipt_value[fx_node_arg.name],
         #    in TorchScript graph.
-        onnx_tensor = fx_name_to_onnxscipt_value[onnx_tensor.name]
-    elif isinstance(onnx_tensor, torch.dtype):
+        return fx_name_to_onnxscipt_value[onnx_tensor.name]
+    if isinstance(onnx_tensor, (tuple, list)) and any(
+        isinstance(ele, torch.fx.Node) and isinstance(ele.meta["val"], torch.SymInt)
+        for ele in onnx_tensor
+    ):
+        # This intends to handle dynamic axes. for example, if the input size of op.Expand
+        # is dynamic, each dimension would be variable (i.e., sym variable in Pytorch
+        # FX graph. Note that sym variable is mapped to tensor in ONNX Script world)
+        # calculated by other operators.
+        sequence_elements = []
+        for tensor in onnx_tensor:
+            if isinstance(tensor, torch.fx.Node) and isinstance(
+                tensor.meta["val"], torch.SymInt
+            ):
+                sequence_elements.append(fx_name_to_onnxscipt_value[tensor.name])
+            else:
+                sequence_elements.append(tensor)
+        # Concat all the elements in the sequence.
+        # shapes are mapped to tensors in ONNX graph (TorchScriptGraph),
+        # so list of sym_ints is concatenated to a tensor before calling ONNX op.
+
+        # For example:
+        #    inputs: [2, 4, TensorA, 1, TensorB]
+        #    outputs: Tensor([Tensor(2), Tensor(4), TensorA, Tensor(1), TensorB])
```

Comment:
```suggestion
        #    outputs: onnx::Concat([Tensor(2), Tensor(4), TensorA, Tensor(1), TensorB])
```
I wonder if this makes more sense.

Reply:
I was kind of referring Tensor in this case as onnx tensor.

Reply:
Is this more accurate? `Tensor([2, 4, TensorA.item(), 1, TensorB.item()])`. I thought about commenting this, but then I wonder if you used `onnx::Concat` to achieve the onnx tensor.

The original line shows nested tensor, which is not a legal representation.

Reply:
How about this:

```python
# For example:
#    inputs: [2, 4, fx.Node(SymIntA), 1, fx.Node(SymIntB)]
#    outputs: op.Concat([op.Constant(2), op.Constant(4), TorchScriptTensor(A), op.Constant(1), TorchScriptTensor(B)])
```